sched/init/nx_start.c

```diff
@@ -727,17 +727,18 @@ void nx_start(void)
 
   /* Announce that the CPU0 IDLE task has started */
 
-  sched_note_start(&g_idletcb[0].cmn);
+  sched_note_start(&g_idletcb[0]);
 
   /* Initialize stdio for the IDLE task of each CPU */
 
   for (i = 0; i < CONFIG_SMP_NCPUS; i++)
     {
+      ttcb = (FAR struct task_tcb_s *)&g_idletcb[i];
       if (i > 0)
         {
           /* Clone stdout, stderr, stdin from the CPU0 IDLE task. */
 
-          DEBUGVERIFY(group_setuptaskfiles(&g_idletcb[i], NULL, true));
+          DEBUGVERIFY(group_setuptaskfiles(ttcb, NULL, true));
```

Comment:
why not cast directly

Reply:
as this turns out the only use of that helper variable, yes we can cast directly in next update.